frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx

```diff
@@ -0,0 +1,127 @@
+  query: Lib.Query;
+  stageIndex: number;
+  onSubmit: (columnsAndSeparators: ColumnAndSeparator[]) => void;
+}
+
+export const CombineColumnsDrill = ({
+  drillInfo,
+  query,
+  stageIndex,
+  onSubmit,
+}: Props) => {
+  const { availableColumns, column, defaultSeparator } = drillInfo;
+  const columnInfo = Lib.displayInfo(query, stageIndex, column);
+  const options = useMemo(() => {
+    return getColumnOptions(query, stageIndex, availableColumns);
+  }, [query, stageIndex, availableColumns]);
+  const [columnsAndSeparators, setColumnsAndSeparators] = useState([
+    getInitialColumnAndSeparator(drillInfo),
+  ]);
+  const [isUsingDefaultSeparator, setIsUsingDefaultSeparator] = useState(true);
+
+  const handleChange = (index: number, change: Partial<ColumnAndSeparator>) => {
+    setColumnsAndSeparators(value => [
+      ...value.slice(0, index),
+      { ...value[index], ...change },
+      ...value.slice(index + 1),
+    ]);
+  };
+
+  const handleAdd = () => {
+    setColumnsAndSeparators(value => [
+      ...value,
+      getNextColumnAndSeparator(drillInfo, options, columnsAndSeparators),
+    ]);
+  };
+
+  const handleRemove = (index: number) => {
+    setColumnsAndSeparators(value => [
+      ...value.slice(0, index),
+      ...value.slice(index + 1),
+    ]);
+  };
+
+  const handleEditSeparators = () => {
+    setIsUsingDefaultSeparator(false);
+  };
+
+  const handleSubmit: FormEventHandler = event => {
+    event.preventDefault();
+    onSubmit(columnsAndSeparators);
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <Card className={styles.card} maw="100vw" miw={340} p="lg">
+        <Title
+          mb="md"
+          order={4}
+        >{jt`Combine ${columnInfo.displayName} with`}</Title>
+
+        <Stack spacing="lg">
+          <Stack spacing="sm">
+            <Stack className={styles.inputs} mah="50vh" spacing="sm">
+              {columnsAndSeparators.map(({ column, separator }, index) => (
+                <ColumnAndSeparatorRow
+                  column={column}
+                  index={index}
+                  key={index}
+                  options={options}
+                  separator={separator}
+                  showLabels={!isUsingDefaultSeparator && index === 0}
+                  showRemove={columnsAndSeparators.length > 1}
+                  showSeparator={!isUsingDefaultSeparator}
+                  onChange={handleChange}
+                  onRemove={handleRemove}
+                />
+              ))}
+            </Stack>
+
+            {isUsingDefaultSeparator && (
+              <Box>
+                <Button p={0} variant="subtle" onClick={handleEditSeparators}>
+                  {jt`Separated by ${formatSeparator(defaultSeparator)}`}
+                </Button>
+              </Box>
+            )}
+          </Stack>
+
+          <Flex align="center" gap="md" justify="space-between">
+            <Button
+              leftIcon={<Icon name="add" />}
```

Comment:
Mantine does not correctly handle it in RTL languages, i.e. the icon will be on the left in RTL layout as well.